src/components/input/inputDate/inputDate.tsx

```diff
@@ -0,0 +1,41 @@
+import classNames from 'classnames';
+import { forwardRef, useRef } from 'react';
+import { mergeRefs } from 'react-merge-refs';
+import { Button } from '../../button';
+import { IconType } from '../../icon';
+import { InputContainer, type IInputComponentProps } from '../inputContainer';
+import { useInputProps } from '../useInputProps';
+
+export interface IInputDateProps extends IInputComponentProps {}
+
+export const InputDate: React.FC<IInputDateProps> = forwardRef((props, ref) => {
+    const { containerProps, inputProps } = useInputProps(props);
+
+    const { className: inputClassName, ...otherInputProps } = inputProps;
+
+    const inputRef = useRef<HTMLInputElement>(null);
+
+    const handleCalendarClick = () => {
+        inputRef.current?.showPicker();
+    };
+
+    return (
+        <InputContainer {...containerProps}>
+            <input
+                type="date"
+                className={classNames('calendar-icon:hidden calendar-icon:appearance-none', inputClassName)}
```

Comment:
For the warning and critical variants, the color of the value should be changed to `warning-800` and `critical-800` respectively. This only seems to apply to the non default value/placeholder of 'mm/dd/yyyy'.

Also, on Firefox both icons are shown (see comment on the TW config file), and when the input is disabled, the `InputContainer` background color needs to be set to the same as the actual date input. Might be that the `InputContainer` component disabled color itself needs to be updated, not sure.

Reply:
- I updated the input-container to set the correct color for values depending on the variant ✅
- Fixed the disabled input style on Firefox ✅
- Time and Date input placeholders will currently use the same color as the value, will discuss this with designer ⌛